internal/provisioner/kops_utils.go

```diff
@@ -110,3 +111,52 @@ func getLoadBalancerEndpoint(ctx context.Context, namespace string, logger log.F
 		}
 	}
 }
+
+// GetNGINXLoadBalancerEndpoint returns the load balancer endpoint of the NGINX service.
+func (provisioner *KopsProvisioner) GetNGINXLoadBalancerEndpoint(cluster *model.Cluster, namespace string) (string, error) {
+	logger := provisioner.logger.WithFields(log.Fields{
+		"cluster":         cluster.ID,
+		"nginx-namespace": namespace,
+	})
+	kops, err := kops.New(provisioner.s3StateStore, logger)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create kops wrapper")
+	}
+	defer kops.Close()
+
+	kopsMetadata, err := model.NewKopsMetadata(cluster.ProvisionerMetadata)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse provisioner metadata")
+	}
+
+	if kopsMetadata.Name == "" {
+		logger.Infof("Cluster %s has no name, assuming cluster installation never existed.", cluster.ID)
+		return "", nil
+	}
+
+	err = kops.ExportKubecfg(kopsMetadata.Name)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to export kubecfg")
+	}
+
+	k8sClient, err := k8s.New(kops.GetKubeConfigPath(), logger)
+	if err != nil {
+		return "", err
+	}
+
+	services, err := k8sClient.Clientset.CoreV1().Services(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+	for _, service := range services.Items {
+		if service.Status.LoadBalancer.Ingress != nil {
+			endpoint := service.Status.LoadBalancer.Ingress[0].Hostname
```

Comment:
How about removing the extra allocation?
```
if len(service.Status.LoadBalancer.Ingres) > 0 {
	endpoint := &service.Status.LoadBalancer.Ingress[0].Hostname
	if *endpoint == "" {
		return "", errors.New("loadbalancer endpoint value is empty")
	}
	return *endpoint, nil
}
```
or just
```
for _, service := range services.Items {
        if len(service.Status.LoadBalancer.Ingres) > 0 && service.Status.LoadBalancer.Ingress[0].Hostname != "" {
	         return service.Status.LoadBalancer.Ingress[0].Hostname, nil	
        }
        return "", errors.New("loadbalancer endpoint value is empty") 
}
```

Reply:
thanks, done :)